src/components/Timer.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+
+import Button from "./Button";
+
+import useInterval from "../hooks/useInterval";
+
+import formatTime from "../utils/formatTime";
+import minutesToSeconds from "../utils/minutesToSeconds";
+
+function Timer() {
+  const [seconds, setSeconds] = useState(minutesToSeconds(25));
+  const [isTimerActive, setIsTimerActive] = useState(false);
+
+  const pauseTimer = () => setIsTimerActive(false);
+  const startTimer = () => setIsTimerActive(true);
+
+  useInterval(() => {
+    if (isTimerActive && seconds) {
+      setSeconds(seconds - 1);
+    }
+  }, 1000);
```

Comment:
Burada şöyle bir sorun var. Aslında `interval` sürekli çalışmaya devam ediyor. Timer'ı durdurduğunda bile her saniyede bir çalışıp, koşulunu kontrol ediyor. Burada `isTimerActive` false ise `interval` ı durdurman gerekir.

Reply:
useInterval'daki (delay !== null) koşulunu kullanabilmek için bir ekleme yaptım. Yanlış anlamadıysam şimdi bu sorunu çözüyor olması lazım.